BoundaryMeshMapping.cpp

```diff
@@ -0,0 +1,279 @@
+        // Grab current position of volumetric mesh.
+        std::vector<dof_id_type> X_dof_indices, X_bdry_dof_indices;
+        for (int d = 0; d < NDIM; ++d)
+        {
+            X_dof_map.dof_indices(d_vol_meshes[part]->node_ptr(vol_node_id), X_dof_indices, d);
+            X_bdry_dof_map.dof_indices(node, X_bdry_dof_indices, d);
+            X_bdry_vec->set(X_bdry_dof_indices[0], (*X_vec)(X_dof_indices[0]));
+            dX_bdry_vec->set(X_bdry_dof_indices[0], (*X_vec)(X_dof_indices[0]) - (*node)(d));
+        }
+    }
+    X_bdry_vec->close();
+    dX_bdry_vec->close();
+    X_bdry_sys.update();
+    dX_bdry_sys.update();
+
+    // We should update the fe_data_manager's elem-patch map
+    d_bdry_data_managers[part]->reinitElementMappings();
+    return;
+}
+
+void
+BoundaryMeshMapping::setInitialConditions()
+{
+    plog << "Setting initial conditions\n";
+    for (unsigned int part = 0; part < d_bdry_meshes.size(); ++part)
+    {
+        // Boundary moves according to what's in the velocity systems
+        EquationSystems* eq_sys = d_bdry_eq_sys_vec[part].get();
+        auto& W_sys = eq_sys->get_system(d_W_sys_name);
+        const DofMap& W_dof_map = W_sys.get_dof_map();
+        NumericVector<double>* W_vec = W_sys.solution.get();
+        // Loop over nodes
+        auto it = d_bdry_meshes[part]->local_nodes_begin();
+        const auto it_end = d_bdry_meshes[part]->local_nodes_end();
+        for (; it != it_end; ++it)
+        {
+            const Node* const node = *it;
+            std::vector<dof_id_type> W_dofs;
+            W_dof_map.dof_indices(node, W_dofs);
+            for (int d = 0; d < NDIM; ++d) W_vec->set(W_dofs[d], 1.0);
+        }
+        W_vec->close();
+        W_sys.update();
+    }
+}
+
+void
+BoundaryMeshMapping::initializeEquationSystems()
+{
+    const bool from_restart = RestartManager::getManager()->isFromRestart();
+    if (from_restart) return;
+    d_hierarchy = d_vol_data_managers[0]->getPatchHierarchy();
+    for (unsigned int part = 0; part < d_bdry_meshes.size(); ++part)
+    {
+        d_bdry_eq_sys_vec[part]->init();
+        d_bdry_data_managers.push_back(
+            FEDataManager::getManager(d_fe_data[part],
+                                      d_object_name + "::FEDataManager::" + std::to_string(part),
+                                      d_input_db,
+                                      d_hierarchy->getFinestLevelNumber() + 1,
+                                      d_vol_data_managers[part]->getDefaultInterpSpec(),
+                                      d_vol_data_managers[part]->getDefaultSpreadSpec(),
+                                      FEDataManager::WorkloadSpec()));
+        d_bdry_data_managers[part]->COORDINATES_SYSTEM_NAME = d_coords_sys_name;
+        d_bdry_data_managers[part]->setPatchHierarchy(d_hierarchy);
+    }
+    d_input_db.setNull();
+    updateBoundaryLocation(0.0, 0.0, true);
+    return;
+}
+
+void
+BoundaryMeshMapping::spreadWallSites(const int w_idx)
```

Comment:
This function actually spreads the wall sites from the boundary mesh to the Eulerian grid.